src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,48 +1,105 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_CAPACITY = 10;
+    private static final double FACTOR_GROW = 1.5;
+    private T[] elementData;
+    private int size;
+
+    public ArrayList() {
+        elementData = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
     @Override
     public void add(T value) {
-
+        if (size == elementData.length) {
+            expandCapacity();
+        }
```

Comment:
```suggestion
        ensureCapacity();
```

Reply:
Where did You get "ensureCapacity()" method ? I don't have such method.
And if I delete checking "if (size == elementData.length)" I will get test failed "checkingResize
<img width="1440" alt="Снимок экрана 2023-07-05 в 21 44 52" src="https://github.com/mate-academy/jv-array-list/assets/115059948/d135e54c-bd46-41b0-9b6f-227b25fe2100">
"

That is why I think it will be better to leave checking size.